src/web/lib/braze/LocalMessageCache.ts

```diff
@@ -0,0 +1,76 @@
+import type appboy from '@braze/web-sdk-core';
+import { SlotName } from './BrazeMessages';
+
+const localStorageKeyBase = 'gu.brazeMessageCache';
+
+type Message = appboy.InAppMessage;
+
+interface Store {
+	getItem: (key: string) => string | null;
+	setItem: (key: string, value: string) => void;
+	removeItem: (key: string) => void;
+}
+
+const keyFromSlotName = (slotName: SlotName): string =>
+	`${localStorageKeyBase}.${slotName}`;
```

Comment:
Would this be simpler as a single queue of braze messages where each stores the slot, rather than one queue per slot? The interface would be identical (we could filter on the slot) but the storage would map more closely to how Braze sees these messages.

I'm very happy with it as-is, just wanted to check what you think. If we ever want to support a Braze message that is flexible in which slot it uses we'd be much better off with a single queue, but as it stands I think it's 6 and a half-dozen. This slot flexibility is purely hypothetical for now and I think it's unlikely we'll do that, so the current choice looks good.

Reply:
To me keeping things separate felt simpler, since at the moment the slots can be blissfully ignorant of what's going on with the other. It's true, I can imagine a future world where a message is agnostic about it's slot, or at least is flexible about where it goes, but that feels like a long way off at the moment.